File: Frontend/src/app/Services/band-concert.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { BandConcert } from '../Class/BandConcert';

import { api_url } from '../Utils/API_URL';

const apiUrl = api_url;

@Injectable({
  providedIn: 'root',
})
export class BandConcertService {
  constructor(private http: HttpClient) {}

  getById(id: number): Observable<any> {
    return this.http.get(`${apiUrl}/bandconcert/${id}`);
  }
  getByBandId(id: number): Observable<any> {
    return this.http.get(`${apiUrl}/bandconcert/band/${id}`);
  }
  getByConcertId(id: number): Observable<any> {
    return this.http.get(`${apiUrl}/bandconcert/concert/${id}`);
  }

  list(): Observable<any> {
    return this.http.get(`${apiUrl}/bandconcert`);
  }

  create(bandConcert: BandConcert): Observable<any> {
    return this.http.post(`${apiUrl}/admin/bandconcert`, bandConcert);
  }

  edit(bandConcert: BandConcert): Observable<any> {
    return this.http.put(`${apiUrl}/admin/bandconcert`, bandConcert);
  }

  delete(id: number): Observable<any> {
    return this.http.delete(`${apiUrl}/admin/bandconcert/${id}`);
  }
}

File: Backend/src/main/java/musiccube/services/bandconcert/BandConcertService.java
package musiccube.services.bandconcert;

import musiccube.entities.Band;
import musiccube.entities.BandConcert;
import musiccube.entities.Concert;

import java.util.Optional;

public interface BandConcertService {

    Optional<BandConcert> getById(int id);
    Iterable<BandConcert> getAll();
    BandConcert save(BandConcert BandConcert);
    void delete(int id);

    Iterable<BandConcert> getByBandId(int bandId);
    Iterable<BandConcert> getByConcertId(int concertId);
}
